common/src/main/java/org/mvndaemon/mvnd/common/Environment.java

```diff
@@ -224,7 +224,7 @@ public enum Environment {
             null,
             "io.takari.maven:takari-smart-builder",
             OptionType.STRING,
-            Flags.OPTIONAL),
+            Flags.DISCRIMINATING | Flags.OPTIONAL),
```

Comment:
I don't think this one is discriminating.  What is discriminating is the list of extensions to load which may actually be affected by this property, but not necessarily.  If you add an exclusion which is not supposed to be loaded, this won't affect the result.  In addition, that property is not really interesting on the daemon side, as the extensions list is loaded by the client...

Reply:
removed